pkg/flow/adapter/dataweavetransformation/adapter.go

```diff
@@ -96,73 +118,104 @@ func (a *dataweaveTransformAdapter) Start(ctx context.Context) error {
-		}
-	case "application/xml":
-		if !isValidXML(event.Data()) {
-			return a.replier.Error(&event, targetce.ErrorCodeRequestParsing,
-				errors.New("invalid XML"), nil)
-		}
-		tmpfile, err = ioutil.TempFile(path, "*.xml")
-		if err != nil {
-			return a.replier.Error(&event, targetce.ErrorCodeAdapterProcess, err, "creating the xml file")
-		}
-	default:
-		return a.replier.Error(&event, targetce.ErrorCodeRequestParsing,
-			errors.New("unexpected type for the incoming event"), nil)
+	err = validateContentType(event.DataContentType(), event.Data())
+	if err != nil {
+		return a.replier.Error(&event, targetce.ErrorCodeRequestValidation,
+			errors.New(err.Error()), nil)
 	}
 
-	errs := registerAndPopulateSpell(a.spell, dwFolder)
-	if errs != nil {
-		return a.replier.Error(&event, targetce.ErrorCodeAdapterProcess, err, "creating the spell")
-	}
+	inputData := event.Data()
+	spell := a.defaultSpell
+	inputContentType := a.defaultInputContentType
+	outputContentType := a.defaultOutputContentType
 
-	if _, err := tmpfile.Write(event.Data()); err != nil {
-		return a.replier.Error(&event, targetce.ErrorCodeAdapterProcess, err, "writing to the file")
+	req := &DataWeaveTransformationStructuredRequest{}
+	if err := event.DataAs(req); err != nil {
+		return a.replier.Error(&event, targetce.ErrorCodeRequestParsing, err, nil)
 	}
 
-	out, err := exec.Command("dw", "-i", "payload", tmpfile.Name(), "--local-spell", dwFolder).Output()
-	if err != nil {
-		return a.replier.Error(&event, targetce.ErrorCodeAdapterProcess, err, "executing the spell")
+	if !a.spellOverride && req.Spell != "" {
+		return a.replier.Error(&event, targetce.ErrorCodeRequestValidation,
+			errors.New("it is not allowed to override Spell per CloudEvent"), nil)
 	}
 
-	err = os.Remove(tmpfile.Name())
-	if err != nil {
-		return a.replier.Error(&event, targetce.ErrorCodeAdapterProcess, err, "removing the file")
+	if req.InputContentType != "" || req.OutputContentType != "" || req.Spell != "" {
+		if req.InputData == "" {
+			return a.replier.Error(&event, targetce.ErrorCodeRequestValidation,
+				errors.New("inputData not found"), nil)
+		}
 	}
 
-	err = os.RemoveAll(dwFolder)
-	if err != nil {
-		return a.replier.Error(&event, targetce.ErrorCodeAdapterProcess, err, "removing the folder")
-	}
+	if a.spellOverride {
+		if req.InputContentType != "" {
+			inputContentType = &req.InputContentType
+		}
 
-	cleaned := bytes.Trim(out, "Running local spell")
-	if err := event.SetData(a.outputContentType, cleaned); err != nil {
-		return a.replier.Error(&event, targetce.ErrorCodeAdapterProcess, err, nil)
-	}
+		if req.OutputContentType != "" {
+			outputContentType = &req.OutputContentType
+		}
 
-	event.SetType(event.Type() + ".response")
-	a.logger.Infof("responding with transformed event: %v", event.Type())
-	if a.sink != "" {
-		if result := a.ceClient.Send(ctx, event); !cloudevents.IsACK(result) {
-			a.logger.Errorf("Error sending event to sink: %v", result)
+		if req.InputData != "" {
```

Comment:
Is there a time when the input data is optional?

Reply:
Well it's not optional if the others parameters are sent, but I'm just checking every parameter before to set it.